### PR TITLE
feat(commerce): mark v2 listings as non-v2 and non-v2 listings as old

### PR DESCRIPTION
**Pull request by Spuffynism**
This is somewhat of an overdue rename. Generally, I've renamed "ProductListingV2" to "ProductListing", and the v1 listings from "ProductListing" to "OldProductListing".

The renamed classes or file names do not affect the API surface of neither the product-listing sub-package, or the commerce sub-package. I've however deleted the v2 facet analytics actions, and removed them from the `product-listing-actions-loader.ts` action loader, which is a breaking change. I don't expect anyone to have been using these though.

[CAPI-763]

[CAPI-763]: https://coveord.atlassian.net/browse/CAPI-763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/api/analytics/commerce-analytics.ts [hunk 1]
```diff
@@ -6,21 +6,21 @@ import {
 import {SearchEventRequest} from 'coveo.analytics/dist/definitions/events';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/api/analytics/commerce-analytics.ts:6]: In fact I think we could delete this file altogether!

File: packages/headless/src/features/commerce/product-listing/product-listing-actions-loader.ts [hunk 2]
```diff
@@ -55,86 +43,3 @@ export function loadProductListingActions(
     fetchProductListing,
   };
 }
-
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/commerce/product-listing/product-listing-actions-loader.ts:58]: 👏

File: packages/headless/src/features/analytics/analytics-utils.ts [hunk 3]
```diff
@@ -145,9 +145,6 @@ export type ProductRecommendationAction =
 export type ProductListingAction =
   PreparableAnalyticsAction<StateNeededByProductListingAnalyticsProvider>;
 
-export type ProductListingV2Action =
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/analytics/analytics-utils.ts:148]: We can also delete `makeCommerceAnalyticsAction`; it was only used in facet-set-product-listing-v2-analytics-actions.ts, which you're deleting in this PR.

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/api/analytics/commerce-analytics.ts (hunk 1 above)
Comment (Spuffynism) [reply packages/headless/src/api/analytics/commerce-analytics.ts]: Done in https://github.com/coveo/ui-kit/pull/4087/commits/515aa44654a616177653b81947a224d210dd263a

**Pull request merged**